### ISSUE TITLE
Project manager crashes when opening in 4.1.dev4 due to OpenGL regression (Windows, AMD)

### ISSUE THREAD

**Issue by xlljc** [godotengine/godot#77760]
### Godot version

Godot_4.1.dev4

### System information

windows10

### Issue description

I tried to open godot_4.1.dev4 on my windows10 machine and found it crashed. The godot console did not have any valid error messages. I have two Windows 10 machines and both crashed

> p1 configuration:
> AMD Ryzen 7 5800X
> 32.0 GB
> AMD Radeon RX6600

> pc2 configuration:
> AMD Ryzen 7 5800H
> 32.0 GB
> NVIDIA GeForce RTX 3070 Laptop

### Steps to reproduce

Run editor

### Minimal reproduction project

none

**Comment by xlljc** [comment]
![2023-06-02_122334](https://github.com/godotengine/godot/assets/63633253/743a3665-d02a-42ac-a5f5-dabe6e9cd9e6)
![gif](https://github.com/godotengine/godot/assets/63633253/a1b11b9a-c1dd-43bd-83e6-27e430e6aaa7)

**Comment by miketgv** [comment]
same issue on my side

**Comment by bitsawer** [comment]
Do any previous Godot versions work? Do you think this is a regression or is this the first time you are trying out Godot?

**Comment by xlljc** [comment]
> Do any previous Godot versions work? Do you think this is a regression or is this the first time you are trying out Godot?

I've been trying the latest versions, from dev1 to dev4, and the first three are all working

**Comment by Zireael07** [comment]
Looks like a regression then

**Comment by xlljc** [comment]
> same issue on my side

I found that I couldn't open the project list, but I was able to just drag the. godot file into godot.exe and the editor worked fine

**Comment by bitsawer** [comment]
> godot file into godot.exe and the editor worked fine

>crashes while recompiling gles3 cached shaders

The Godot project manager uses OpenGL renderer by default, so if things work better by running the editor directly (and the actual project uses Forward Plus or Mobile renderer) it might have something to do with OpenGL renderer.

You can try to bypass the project manager and edit project directly by drag-and-dropping or using `--editor` and `--path` command line arguments and see if the project editor starts directly: https://docs.godotengine.org/en/stable/tutorials/editor/command_line_tutorial.html Or maybe even test with different `--rendering-method` command line arguments. See also check out `--help`.

**Comment by miketgv** [comment]
Drag and drop worked for me also

**Comment by miketgv** [comment]
I made it work with  --rendering-driver vulkan.
you can create a shortcut to Godot_v4.1-dev4_mono_win64.exe and in Properties add --rendering-driver vulkan => functioning Project Manager

**Comment by Calinou** [comment]
This may be a regression from https://github.com/godotengine/godot/pull/76092, even though https://github.com/godotengine/godot/pull/77163 was already merged in 4.1.dev4.

**Comment by miketgv** [comment]
> This may be a regression from #76092, even though #77163 was already merged in 4.1.dev4.
> 
> Can you reproduce this issue in 4.0.3? Also, can you reproduce the issue in 4.1.dev4 _without_ Mono?

I've just tried 4.1dev4 without mono and behaves the same (crashes on normal run, works with --rendering-driver vulkan)
I'm currently working on a project on 4.0.3 and works as expected

**Comment by akien-mga** [comment]
> This may be a regression from #76092, even though #77163 was already merged in 4.1.dev4.

Both were already merged in 4.1.dev3, so if the regression happened between dev3 and dev4, it should be a different PR.

This is more likely to be caused by one of these changes to `drivers/gles3/` between dev3 and dev4:
```
commit f2e91eae639406a5fccea0210042a3e79760d502
Merge: aa739bae4f 391a1bf60e
Author: Rémi Verschelde <rverschelde@gmail.com>
Date:   Mon May 29 10:29:08 2023 +0200

    Merge pull request #77224 from patrickdown/implement_texture_2d_layered_initialize
    
    Implemented texture_2D_layered_initialize

commit 391a1bf60e757c59cfd32e9e1dd851b57cc7c1d1
Author: patrickdown <patrick.down@gmail.com>
Date:   Thu May 18 11:41:17 2023 -0400

    Implemented texture_2D_layered_initialize

commit f6dcd7f51a65f9fbc4c98583016bc85d130b3106
Merge: 08fcf27038 43b0702a02
Author: Yuri Sizov <11782833+YuriSizov@users.noreply.github.com>
Date:   Sat May 27 17:28:29 2023 +0200

    Merge pull request #75181 from dalexeev/fix-light-only-mode
    
    Fix "Light Only" mode of `CanvasItemMaterial`

commit 6d3634e29f32d1d481a2d71e3eb7cc0718bde0a9
Author: Dani Frank <pdaniq@gmail.com>
Date:   Sat Mar 25 12:46:33 2023 +0300

    fix gles texture uniform array binding

commit 43b0702a0266a823ae42d0c048a6d6baeaceaa06
Author: Danil Alexeev <danil@alexeev.xyz>
Date:   Tue Mar 21 12:18:15 2023 +0300

    Fix "Light Only" mode of `CanvasItemMaterial`
```

**Comment by BastiaanOlij** [comment]
@akien-mga sadly I don't have a mono setup here (really should rectify this) but a normal master build runs fine so it has to be a mono specific regression.

Having a look at the commits you mentioned, nothing stands out. Anything in `391a1bf60e757c59cfd32e9e1dd851b57cc7c1d1` would have resulted in console output, `43b0702a0266a823ae42d0c048a6d6baeaceaa06` doesn't seem to structurally change anything, I'm not entirely sure what `6d3634e29f32d1d481a2d71e3eb7cc0718bde0a9` is about but nothing stands out that would cause mono to panic.

We have had issues in the past with parts of the system calling into the mono system before mono is ready, if we're logging errors early that could explain things, but I would expect those errors to be logged in a normal build too.

I'll see if I can get a mono setup working tomorrow.

**Comment by YuriSizov** [comment]
@BastiaanOlij The report above suggests that the issue is reproduceable without mono as well. Although...

@miketgv Do you use the [self-contained mode](https://docs.godotengine.org/en/stable/tutorials/io/data_paths.html#self-contained-mode)? Can you try each version (with mono, without mono) in that mode?

**Comment by stilestr** [comment]
Chiming in as another AMD user with the same issue, although did not try Mono, just regular version.

Launching it immediately gives a grey screen, then crashes. All previous versions work fine, and nothing about my system has changed.
![image](https://github.com/godotengine/godot/assets/4535064/68bb98c7-e497-42d3-b186-31ea114146c4)

Specs:

- AMD Ryzen 7 5800x
- AMD Radeon RX 5700 XT
- 16Gigs of ram
- Windows 11 Pro 64-bit

Setting the rendering driver to Vulkan in the launch parameters resolves the issue, same as above posters.

**Comment by BSSM7** [comment]
same issue here. (windows 10, AMD ryzen 5900x, Radeon 6950xt ,driver: 23.5.2 ==> last driver).

- I tried earlier versions and worked well.
- I tried this version (Godot 4.1 dev4) on **linux** with this pc and it works fine.
- I tried this version on my laptop (intel core i7 + intel iris xe) and it works fine.

maybe AMD issue.



https://github.com/godotengine/godot/assets/121864036/c6f16ce4-a525-4d1d-ada3-84b1da72c0b6

**Comment by Calinou** [comment]
OP has mentioned that it crashes on a NVIDIA setup, although they may have used the AMD integrated graphics if they overrode the selection (Godot requests the dedicated GPU by default).

**Comment by RedworkDE** [comment]
To help narrow down the issue, could someone with this issue go though these builds and report which cause a crash and which start into the project manager without problems: https://mega.nz/folder/AjYRiTga#jyCIWuI_MW1HOic89VEozg

**Comment by stilestr** [comment]
> To help narrow down the issue, could someone with this issue go though these builds and report which cause a crash and which start into the project manager without problems: https://mega.nz/folder/AjYRiTga#jyCIWuI_MW1HOic89VEozg

1. f6dcd7f51a65f9fbc4c98583016bc85d130b3106
2. 6d3634e29f32d1d481a2d71e3eb7cc0718bde0a9
3. 43b0702a0266a823ae42d0c048a6d6baeaceaa06
4. a67d37f7cffe7f31c68e971280950d487ea99e2c

All of these worked.

**Comment by akien-mga** [comment]
Maybe related to https://github.com/godotengine/godot/pull/73450.

**Comment by BastiaanOlij** [comment]
Reading up on this, yeah must be AMD related then, for me the same thing, testing on NVidia I have no issues.

Has anyone been able to run this on a full debug build with `--gpu-validation` and `--verbose` switches?

**Comment by RedworkDE** [comment]
> Maybe related to #73450.

Unlikely, a round of remote bisection (https://github.com/godotengine/godot/issues/77760#issuecomment-1574231922) points at #77224 tho it hasn't been confirmed yet

(Again if someone with this issue could please run these https://github.com/godotengine/godot/issues/77760#issuecomment-1574259301 and report back if they crash or open the project manager)

Either way the currently remaining commits that may have caused this are:
```
f2e91eae639406a5fccea0210042a3e79760d502 (refs/bisect/bad) Merge pull request #77224 from patrickdown/implement_texture_2d_layered_initialize
aa739bae4f84fc2b7a57692896aad29b843b07fa Merge pull request #76616 from TokageItLab/implement-anim-node-sub2
c97201babcb76e6fb0e7558074dfe7808982eb00 Merge pull request #76545 from JoNax97/gradient_color_spaces
990943782a80e47b4a46587c298a338e9a2c119c Merge pull request #74959 from MewPurPur/nice-curves
9f05e16727a6776c0bf27c18d5ce0d016542494a Merge pull request #70940 from MewPurPur/better-gradient2d-editor
6dcd490d03ac98b9759ceac1dddbe9b67f6ea74a Merge pull request #66763 from MewPurPur/remove-bookmark-hiding
a3c4a4b039e9c0c5947744bbaf7f9a9bc123982c Overhaul the Curve Editor
28cca66d2cfe25d6d7bf5a2a26ab7bd366029669 Merge pull request #77541 from TokageItLab/remove-outdated-docs-from-skeleton
06ccbfe8eec6e840db3df2360dc2804386f940b8 (HEAD) Merge pull request #77534 from aaronfranke/gltf-node-index
c670c4334a6901410128f8b66a0ce7d097e66e5d Merge pull request #65902 from MJacred/editor/copysysteminfo
391a1bf60e757c59cfd32e9e1dd851b57cc7c1d1 Implemented texture_2D_layered_initialize
5e139c2ab927f0546a5f6b6a974a931b53013978 Add a get_node_index method to GLTFState
9e5bf3d589fec3f342231fd9e0a16b03b83469d3 Copy system info to clipboard + Update bug_report.yml
9a3221f1edb057d8ec3dfe55f9966d3ae5c8acb1 Merge pull request #77419 from MewPurPur/remove-unused-svgs
36e7800679e406ac79e6865cabaf0e7e69bc81d8 Merge pull request #77511 from KoBeWi/destroy_size()
c87dc07fdd52de4437f828b26eb0c958168efdb0 Merge pull request #77514 from bruvzg/fix_dup_breaks
9128ed47198630026737366d788a4a2478853750 Merge pull request #77517 from KoBeWi/TranslateThisRightNow
26cc4bbd82e991de7fdd4a44cadcbc1558e16205 Remove out-dated lines from class reference of Skeleton3D
c9c8205644253e5aea712dfd893a39499fc773de Implement NodeSub2 to AnimationTree & improve mathematical Nodes
c07b2fcf4dd9bc60f6d7ead81b8de8ee19178729 Add Linear SRGB and OKLab color spaces to Gradient.
adf427c806b902123e722e86fda55aa173ec5aa2 Translate inspector changes indicator
8a4a883854d24f864ab39be1a152f58d754db91e [TextServer] Prevent duplicate line breaks on virtual spaces when line width is significantly smaller than character width.
05663b8633f470ad4baefa896fb9a14596bf5d93 Minor improvements to sub-resource list
860f51b0503879c72553736f65ad15f4b45387e7 Remove unused SVG icons
39b79bbd1e83d1ac46a1dacdd17cd4b05a0000de Improvements to Gradient2D Editor
e0e236edfc75cb4d9999d619edac3cec8e3ea323 Remove editor setting for hiding bookmarks
```

**Comment by RedworkDE** [comment]
Thanks, which confirms this to be a regression from #77224, as one of them was master with just that PR reverted.

**Comment by clayjohn** [comment]
Has anyone reproduced the bug on a debug build and can post a stack trace?

**Comment by FranckRJ** [comment]
I've got this but I don't know if it's useful:

	Exception thrown at 0x00007FF8BA27236A (atio6axx.dll) in godot.windows.editor.x86_64.exe: 0xC0000005: Access violation reading location 0xFFFFFFFFFFFFFFFF.

	atio6axx.dll!00007ff8ba27236a()	Unknown
 	atio6axx.dll!00007ff8ba272a06()	Unknown
 	atio6axx.dll!00007ff8ba271f75()	Unknown
 	atio6axx.dll!00007ff8ba1e5d3e()	Unknown
 	atio6axx.dll!00007ff8ba3208fe()	Unknown
 	atio6axx.dll!00007ff8b9efb79e()	Unknown
 	atio6axx.dll!00007ff8b9e8c4d0()	Unknown
 	atio6axx.dll!00007ff8b9e8e488()	Unknown
 	atio6axx.dll!00007ff8b9e4a4ce()	Unknown
 	atio6axx.dll!00007ff8b99e6991()	Unknown
 	atio6axx.dll!00007ff8b9dd46ed()	Unknown
 	atio6axx.dll!00007ff8b9d65f36()	Unknown
 	atio6axx.dll!00007ff8b9d5ce58()	Unknown
 	atio6axx.dll!00007ff8b9aca84f()	Unknown
 	atio6axx.dll!00007ff8bb7f5096()	Unknown
 	kernel32.dll!00007ff9780b26ad()	Unknown
 	ntdll.dll!00007ff978f8a9f8()	Unknown

**Comment by FranckRJ** [comment]
> Reading up on this, yeah must be AMD related then, for me the same thing, testing on NVidia I have no issues.
> 
> Has anyone been able to run this on a full debug build with `--gpu-validation` and `--verbose` switches?

It shows this:

```
$ ./godot.windows.editor.x86_64.exe --gpu-validation --verbose
Godot Engine v4.1.dev.custom_build.db5113de3 - https://godotengine.org
TextServer: Added interface "Dummy"
TextServer: Added interface "ICU / HarfBuzz / Graphite (Built-in)"
Using "winink" pen tablet driver...
Shader 'CanvasSdfShaderGLES3' SHA256: 72cda41890bce2f09217c172023d31e735a3fee605bad83c8e63f1a981a36a77
Shader 'SkeletonShaderGLES3' SHA256: add3f03e6eebdffdf0073fae42195857307eddada333b82998642329511917a2
Segmentation fault
```

**Comment by BSSM7** [comment]
> Reading up on this, yeah must be AMD related then, for me the same thing, testing on NVidia I have no issues.
> 
> Has anyone been able to run this on a full debug build with `--gpu-validation` and `--verbose` switches?

It shows this:

`Godot Engine v4.1.dev4.official.5c2295ff5 - https://godotengine.org
TextServer: Added interface "Dummy"
TextServer: Added interface "ICU / HarfBuzz / Graphite (Built-in)"
Using "winink" pen tablet driver...
Shader 'CanvasSdfShaderGLES3' SHA256: 72cda41890bce2f09217c172023d31e735a3fee605bad83c8e63f1a981a36a77
Shader 'SkeletonShaderGLES3' SHA256: add3f03e6eebdffdf0073fae42195857307eddada333b82998642329511917a2
Shader 'ParticlesShaderGLES3' SHA256: 9bf432d37b0b4792d67aa9f577258845d4047e3f98e990aeca6fd20a868c4db2
Shader 'ParticlesCopyShaderGLES3' SHA256: c778843b0cd1c7ce4c621fa2b924f703ac4001faef0555114b81a76c264a6908
Shader 'CopyShaderGLES3' SHA256: b1bc5a355ed9bda23f291956f121e0535b4e81b54cdac398412d200e6133161f
Shader 'CanvasShaderGLES3' SHA256: 007cddfcf35f7d57b219cffec53f6a7d9fe7a1979be4db64c6450cfac6000053
Shader 'CanvasOcclusionShaderGLES3' SHA256: 968915a469db9a7d505be274e8060c8288c15a5865baeb8362b3b0e2cacb16f3
Shader 'SceneShaderGLES3' SHA256: 7978059ed235406c8eb185f813778475d9fb0316458c2782764a5e2dd98e930d
Shader 'SkyShaderGLES3' SHA256: 1fe105946e63973aecf5198091930c2bdb17778012ef0253e325ade3a297e6c8
Shader 'CubemapFilterShaderGLES3' SHA256: 2b651c285a02b7d4221af1827e1613c03e73e600ebc40b9e6a0b22bb855bd75f
OpenGL API 3.3.0 Core Profile Context 23.5.2.230523 - Compatibility - Using Device: ATI Technologies Inc. - AMD Radeon RX 6950 XT
WASAPI: Activated output_`

**Comment by antimundo** [comment]
I think I have the same issue, the Project Manager just doesn't work when I open it.
- Godot 4.1-beta1 official
- Windows 10 22H2
- i5 7600k
- 16gb ram DDR4
- AMD RX 6800 16gb

https://github.com/godotengine/godot/assets/64472605/fef3b532-ba0d-45ca-a7ff-1a1ca73deea4

**Comment by Calinou** [comment]
As mentioned in https://godotengine.org/article/dev-snapshot-godot-4-1-beta-1/#known-issues, the fix is only merged in the upcoming 4.1.beta2, **not** 4.1.beta1. There's a workaround listed in the link.
